cfdm/data/subarray/mixin/linearinterpolation.py

```diff
@@ -0,0 +1,46 @@
+class LinearInterpolation:
+    """Mixin class for linear interpolation.
+
+    See CF appendix J "Coordinate Interpolation Methods".
+
+    .. versionadded:: (cfdm) 1.9.TODO.0
+
+    """
+
+    def _linear_interpolation(self, ua, ub, d1, s=None, returns=False):
+        """Interpolate linearly between two points.
+
+        u = fl(ua, ub, s) = ua + s*(ub-ua)
+                          = ua*(1-s) + ub*s
+
+        See CF appendix J "Coordinate Interpolation Methods".
+
+        .. versionadded:: (cfdm) 1.9.TODO.0
+
+        .. seealso:: `_s`
+
+        :Parameters:
+
+            ua, ub: `numpy.ndarray`
+                The first and second tie points in index space of
+                subsampled dimension 1 (in the sense of CF appendix J
+                Figure J.1).
+
+            {{d1: `int`}}
+
+            {{s: array_like, optional}}
+
+        :Returns:
+
+            `numpy.ndarray`
+                The result of interpolating the tie points to interior
+                locations implied by *s*.
+
+        """
+        s = self._s(d1, s=s)
+        u = ua + s * (ub - ua)
+
+        if returns:
```

Comment:
I was going to request that the `returns` keyword here be documented, but from a quick `grep` it doesn't seem to be used anyway, so just drop it?

Reply:
Sure thing. It was used in an earlier iteration of the code, but has clearly gone by the way side. I'll take it out (https://github.com/NCAS-CMS/cfdm/pull/168/commits/0e8bf3185df7062fdc1ccf5761531b7f3e71498d)